From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Vishnu Mandalapu <vmandalapu@nvidia.com>
Date: Tue, 4 Feb 2025 19:55:12 +0000
Subject: [PATCH 092/240] NVIDIA: SAUCE: drivers: tegra: virt: runtime
 frequency

 - Add support to change VM profiler frequency in runtime.

Jira HYP-19279
Bug 4551265

Signed-off-by: Vishnu Reddy Mandalapu <vmandalapu@nvidia.com>
Change-Id: Ifabbf0ab0f6c025cf3932d51fbffbe2f6a2405a5
Reviewed-on: https://git-master.nvidia.com/r/c/linux-stable/+/3296251
(cherry picked from commit de0d5508bbbc4f574a14239195e2157cc34d2286)
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3305217
Tested-by: Sumit Gupta <sumitg@nvidia.com>
Reviewed-by: Jon Hunter <jonathanh@nvidia.com>
Reviewed-by: Bibek Basu <bbasu@nvidia.com>
(cherry picked from commit dffd783fc499a8c95675f82567b0940636ec8166)

diff --git a/include/soc/tegra/virt/syscalls.h b/include/soc/tegra/virt/syscalls.h
index 0b337ec2165d..bd73c57e85c8 100644
--- a/include/soc/tegra/virt/syscalls.h
+++ b/include/soc/tegra/virt/syscalls.h
@@ -525,7 +525,7 @@ static __attribute__((always_inline)) inline void hyp_call44(uint16_t id, uint64
 __attribute__((no_sanitize_address))
 static inline int hyp_trace_get_mask(uint64_t *value)
 {
-	uint64_t args[4] = { 0U, 0U, 0U, 0U };
+	uint64_t args[4] = { *value, 0U, 0U, 0U };
 
 	hyp_call44(HVC_NR_TRACE_GET_EVENT_MASK, args);
 	if (args[0] == 0U)
@@ -535,9 +535,9 @@ static inline int hyp_trace_get_mask(uint64_t *value)
 }
 
 __attribute__((no_sanitize_address))
-static inline int hyp_trace_set_mask(uint64_t mask)
+static inline int hyp_trace_set_mask(uint64_t type, uint64_t value)
 {
-	uint64_t args[4] = { mask, 0U, 0U, 0U };
+	uint64_t args[4] = { type, value, 0U, 0U };
 
 	hyp_call44(HVC_NR_TRACE_SET_EVENT_MASK, args);
 	return (int) args[0];
-- 
2.55.0

